pathod/test.py

```diff
@@ -39,39 +41,51 @@ def p(self, spec):
         """
         return "%s/p/%s" % (self.urlbase, spec)
 
-    def info(self):
-        """
-            Return some basic info about the remote daemon.
-        """
-        resp = requests.get("%s/api/info" % self.urlbase, verify=False)
-        return resp.json()
-
     def text_log(self):
         return self.logfp.getvalue()
 
+    def wait_for_silence(self, timeout=5):
+        start = time.time()
+        while 1:
+            if time.time() - start >= timeout:
+                raise TimeoutError(
+                    "%s service threads still alive" %
+                    self.thread.server.counter.count
+                )
+            if self.thread.server.counter.count == 0:
+                return
+
+    def expect_log(self, n, timeout=5):
```

Comment:
Yes, I think we still do. First, this is our synchronisation mechanism to ensure that tests don't bleed into each other. Second, some bug classes, like the websocket one fixed in this PR, exhibit themselves through failure to properly exit.